core/src/mindustry/content/Fx.java

```diff
@@ -2445,6 +2445,29 @@ public class Fx{
         Lines.poly(e.x, e.y, 6, e.rotation + e.fin());
     }).followParent(true),
 
+    arcShieldBreak = new Effect(40, e -> {
+        Lines.stroke(3 * e.fout(), e.color);
+        if(e.data instanceof Unit){
+            Unit u = (Unit) e.data;
+            ShieldArcAbility ab = (ShieldArcAbility) Structs.find(u.abilities, a -> a instanceof ShieldArcAbility);
+            if(ab != null){
+                Vec2 pos = Tmp.v1.set(ab.x, ab.y).rotate(u.rotation - 90f).add(u);
+                Lines.arc(pos.x, pos.y, ab.radius + ab.width/2, ab.angle / 360f, u.rotation + ab.angleOffset - ab.angle / 2f);
+                Lines.arc(pos.x, pos.y, ab.radius - ab.width/2, ab.angle / 360f, u.rotation + ab.angleOffset - ab.angle / 2f);
+                for(int i : new int[]{1, -1}){
+                    Seq<Float> lp = new Seq<>();
```

Comment:
Calling `new` in the main loop anywhere isn't acceptable. Use a temporary static variable if you must..

Reply:
Thank you for your guidance!
The methods I have been using that I believe are correct may not be suitable. Thank you again.